src/storage/observable/fromTask.ts

```diff
@@ -0,0 +1,14 @@
+import { storage } from 'firebase/app';
+import 'firebase/storage';
+import { Observable } from 'rxjs/Observable';
+
+export function fromTask(task: storage.UploadTask) {
+  return new Observable<storage.UploadTaskSnapshot | undefined>(subscriber => {
+    task.on('state_changed', 
+      (snap: storage.UploadTaskSnapshot) => subscriber.next(snap),
+      e => subscriber.error(e),
+      () => subscriber.complete()
+    );
+    return { unsubscribe: task.cancel };
```

Comment:
You should just return a function here. Also you're going to want to retain the handler you added above with `task.on('state_changed', handler)` so you can remove it (presumably with `off`) otherwise you could end up with memory leaks.